### PR TITLE
purego: skip importing `runtime/cgo` for Windows

### LINKED ISSUE #39: `go build` a main package with purego with Cgo enabled might fail on Windows and Linux
https://github.com/ebitengine/purego/blob/test/example/main.go

The test bot builds the above main package like below:

```
go build -v ./...
```

And this failed at least on Linux like this:

https://github.com/ebitengine/purego/runs/8276795548

```
# github.com/ebitengine/purego/example
2022/09/09 18:57:24 duplicated definition of symbol _cgo_init
```

I think this is a minimal reproducible case of hajimehoshi/ebiten#2313: just building a main package with purego with Cgo enabled.

### PR DESCRIPTION
**Pull request by hajimehoshi**
This change resolves these issues:

* Buiding purego for `GOOS=darwin` and `GOARCH=arm64` with Go 1.15
  is not supported so far (#37). Skip this.
* Buiding purego for `GOOS=windows` with Cgo failed due to duplicated
  symbols (#39). Let's not import `runtime/cgo` for Windows. `fakecgo` is
  implemented only for Darwin so this should be fine. As a side effect, purego
  can no longer be imported from non-Darwin or non-Windows so far.

Updates #37
Closes #39

### CODE REVIEW

**Comment by hajimehoshi** [comment]
Please take a look, thanks

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
           go-version: ${{ matrix.go }}
 
       - name: go vet
-        if: ${{ !startsWith(matrix.os, 'windows-') && !startsWith(matrix.os, 'macos-') }}
+        if: ${{ !startsWith(matrix.os, 'macos-') && !startsWith(matrix.os, 'windows-') }}
```
Comment (hajimehoshi) [inline .github/workflows/test.yml:32]: This change is not related to the issue. I just wanted to reorder them in an alphabetical order. I can make it in a later PR.

Comment (hajimehoshi) [reply .github/workflows/test.yml]: Reverted this change.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -48,15 +48,21 @@ jobs:
...
-          env GOOS=darwin GOARCH=arm64 go build -v ./...
 
       - name: go build (Windows, ARM64)
         if: ${{ !startsWith(matrix.go, '1.15.') && !startsWith(matrix.go, '1.16.') }}
         run: |
           # Check cross-compiling Windows binaries.
           env GOOS=windows GOARCH=arm64 go build -v ./...
 
+      - name: go build (macOS, ARM64)
+        if: ${{ !startsWith(matrix.go, '1.15.') }}
```
Comment (TotallyGamerJet) [inline .github/workflows/test.yml:59]: Should we add a comment describing why we don't build for 1.15?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -48,15 +48,21 @@ jobs:
 
           # Check cross-compiling macOS binaries.
           env GOOS=darwin GOARCH=amd64 go build -v ./...
-          env GOOS=darwin GOARCH=arm64 go build -v ./...
 
       - name: go build (Windows, ARM64)
         if: ${{ !startsWith(matrix.go, '1.15.') && !startsWith(matrix.go, '1.16.') }}
```
Comment (TotallyGamerJet) [inline .github/workflows/test.yml:53]: Ditto for the comment?

**Review by TotallyGamerJet — COMMENTED** [review]
Thoughts?

**Comment by TotallyGamerJet** [comment]
Why the swap?

File: .github/workflows/test.yml (hunk 2 above)
Comment (hajimehoshi) [reply .github/workflows/test.yml:61]: Done

File: .github/workflows/test.yml (hunk 3 above)
Comment (hajimehoshi) [reply .github/workflows/test.yml:54]: Done

**Comment by hajimehoshi** [comment]
> Why the swap?

I think it's reverted.

**Review by TotallyGamerJet — APPROVED** [review]
LGTM!

**Pull request merged**